src/detect-filename.c

```diff
@@ -99,45 +101,23 @@ void DetectFilenameRegister(void)
     sigmatch_table[DETECT_FILE_NAME].Setup = DetectFilenameSetupSticky;
     sigmatch_table[DETECT_FILE_NAME].flags = SIGMATCH_NOOPT|SIGMATCH_INFO_STICKY_BUFFER;
 
-    // this is required by filestore, and filesize
-    for (int i = 0; file_protos_ts[i].alproto != ALPROTO_UNKNOWN; i++) {
-        DetectAppLayerInspectEngineRegister2("files", file_protos_ts[i].alproto, SIG_FLAG_TOSERVER,
-                file_protos_ts[i].progress, DetectFileInspectGeneric, NULL);
-    }
-    for (int i = 0; file_protos_tc[i].alproto != ALPROTO_UNKNOWN; i++) {
-        DetectAppLayerInspectEngineRegister2("files", file_protos_tc[i].alproto, SIG_FLAG_TOCLIENT,
-                file_protos_tc[i].progress, DetectFileInspectGeneric, NULL);
-    }
-    g_file_match_list_id = DetectBufferTypeGetByName("files");
+    DetectBufferTypeSetDescriptionByName("file.name", "http user agent");
 
-    for (int i = 0; file_protos_ts[i].alproto != ALPROTO_UNKNOWN; i++) {
-        if (file_protos_ts[i].alproto == ALPROTO_HTTP2) {
-            // no filename on HTTP2 files
```

Comment:
@jlucovsky do you know the consequences of this removal ?

Now, HTTP2 filename buffer is registered (even if it will always be empty)

And detect-parse.c has still error on ` "protocol HTTP2 doesn't support file name matching"` for a rule HTTP2 only like `alert http2 any any -> any any (file.name; content:"/a/expl/2008.mp4"; startswith; endswith; sid:24; rev:1;)`

Reply:
if this is a (new) bug then please create a ticket for it

Reply:
Well, I thought at first, but I am not sure this is a bug, since there is the guard anyways in detect-parse.c

Do you have an opinion here ?